### PR TITLE
refactor: use rctx.download with file: URL to copy files in npm_translate_lock

### PR DESCRIPTION
**Pull request by gregmagolan**
With this approach text files & binary files can follow the same code path

### CODE REVIEW

File: npm/private/npm_translate_lock_state.bzl [hunk 1]
```diff
@@ -328,53 +328,24 @@ def _write_action_cache(priv, rctx, label_store):
...
-        _copy_binary_file(priv, rctx, label_store, key)
-
-def _is_text_file(p):
-    # we can assume some files types are text files that are commonly copied
-    return p.endswith(".yaml") or p.endswith(".json") or p.endswith(".npmrc") or p.endswith(".lock")
 
-def _copy_text_file(priv, rctx, label_store, key):
+    # NB: rctx.read will convert binary files to text but that is acceptable for
+    # the purposes of calculating a hash of the file
     contents = rctx.read(label_store.path(key))
```
Comment (jbedard) [inline npm/private/npm_translate_lock_state.bzl:334]: Move this into the `if`?

Comment (gregmagolan) [reply npm/private/npm_translate_lock_state.bzl:334]: the read? it always needs to happen to calculate the hash

Comment (gregmagolan) [reply npm/private/npm_translate_lock_state.bzl]: ohh. I see what you mean.

Comment (gregmagolan) [reply npm/private/npm_translate_lock_state.bzl]: 👍

**Review by jbedard — APPROVED** [review]

**Pull request merged**